Doppler.Sap/Utils/DateTimeProvider.cs

```diff
@@ -1,9 +1,21 @@
 using System;
+using System.Runtime.InteropServices;
 
 namespace Doppler.Sap.Utils
 {
     public class DateTimeProvider : IDateTimeProvider
     {
         public DateTime UtcNow => DateTime.UtcNow;
+
+        public DateTime GetDateByTimezoneId(DateTime date, string timezoneId)
+        {
+            if (!string.IsNullOrEmpty(timezoneId))
+            {
+                var cstZone = TimeZoneInfo.FindSystemTimeZoneById(timezoneId);
+                return TimeZoneInfo.ConvertTimeFromUtc(date, cstZone);
+            }
+
+            return date;
+        }
```

Comment:
I think that it could be done better using DateTimeOffset, but at the moment it seems to be fine.

Did you add tests for this function? (Be sure to test DateTime values with different values in the [Kind property](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.kind?view=net-5.0))

Reply:
👍, I added some unit tests

Reply:
I added more uni test to use predefined values and different kind. Let me know if you want that I add more unit tests.